datafusion/functions/src/datetime/to_date.rs

```diff
@@ -118,3 +134,52 @@ impl ScalarUDFImpl for ToDateFunc {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use arrow::{compute::kernels::cast_utils::Parser, datatypes::Date32Type};
+    use datafusion_common::ScalarValue;
+    use datafusion_expr::{ColumnarValue, ScalarUDFImpl};
+
+    use super::ToDateFunc;
+
+    #[test]
+    fn test_year_9999() {
+        let date_str = "9999-12-31";
```

Comment:
this is a good date to test with

can you also add a test with other interesting dates?

- year 0001 (also common in some data, just like 9999)
- a year before 1970 epoch
- dates during julian/gregorian switch for some western countries (1582-10-10)
- negative years -- these represent problems to some parsing libraries
- 5+-digit year (after 9999) -- these represent problems to some parsing libraries

Reply:
Thanks for the suggestions. I added tests for (almost) all the examples you mentioned. To make negative years work, I had to adapt the code, because currently negative years also do not work.
I intentionally did not add support for 5-digit years, because not even the ISO 8601 standard supports them.

Reply:
i have no issue not supporting 5+-digit years, but interestingly they also can show in real data. imagine some data using 9999-12-31 as a placeholder and then this being converted to a timestamp with time zone and then converted to a different zone... this may produce 10000-01-01. And indeed, at least some query engines do support such dates, but you're right that 9999-12-31 is what they "are obliged" to support.

just saying... it's fine to omit this here for now


>  I added tests for (almost) all the examples you mentioned.


thank you!